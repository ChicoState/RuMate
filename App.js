import React from 'react';
import firebase from 'firebase';
import { StyleSheet } from 'react-native';
import { useScreens } from 'react-native-screens';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import { createBottomTabNavigator } from 'react-navigation-tabs';
import { createStackNavigator } from 'react-navigation-stack';
import Icon from 'react-native-vector-icons/FontAwesome';
/* our imports */
import HomeScreen from './src/screens/HomeScreen';
import TasksScreen from './src/screens/TasksScreen';
import AuthScreen from './src/screens/AuthScreen';
import CreateTaskScreen from './src/screens/CreateTaskScreen';
import AddRoommateScreen from './src/screens/AddRoomateScreen';
import CalendarScreen from './src/screens/CalendarScreen';
import BillsScreen from './src/screens/BillsScreen';
import CreateBillScreen from './src/screens/CreateBillScreen';
import CreateRoommateGroupScreen from './src/screens/CreateRoommateGroupScreen';
import MessageScreen from './src/screens/MessagesScreen';
import CreateConversationScreen from './src/screens/CreateConversationScreen';
import ConversationScreen from './src/screens/ConversationScreen';
import InvitationsScreen from './src/screens/InvitationsScreen';
import AccountScreen from './src/screens/AccountScreen';
import PriorityScreen from './src/screens/PriorityScreen';
import ChangeDetailsScreen from './src/screens/ChangeDetailsScreen';

useScreens();
// ################################ //
// Remove to debug errors/warnings  //
   console.disableYellowBox = true; //
// ################################ //
const firebaseConfig = {
  apiKey: "AIzaSyA_ZWMR-MFG_ZHNK4_WuEeHoLP9vzsY_Vk",
  authDomain: "rumate-faaeb.firebaseapp.com",
  databaseURL: "https://rumate-faaeb.firebaseio.com",
  projectId: "rumate-faaeb",
  storageBucket: "",
  messagingSenderId: "484837108351",
  appId: "1:484837108351:web:53a875e9546d00aa2dad4a"
};
// Initialize Firebase
firebase.initializeApp(firebaseConfig);

const tabNavigator = createBottomTabNavigator({
  Home: {
    screen: HomeScreen,
    navigationOptions: {
      tabBarLabel: "Home",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="home" size={20} />
      )
    },
  },
  Tasks: {
    screen: TasksScreen,
    navigationOptions: {
      tabBarLabel: "Tasks",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="tasks" size={20} />
      )
    },
  },
  Bills: {
    screen: BillsScreen,
    navigationOptions: {
      tabBarLabel: "Bills",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="dollar" size={20} />
      )
    },
  },
  Messages: {
    screen: MessageScreen,
    navigationOptions: {
      tabBarLabel: "Messages",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="comments" size={20} />
      )
    },
  }
},
{
  initialRouteName: 'Home',
});

const taskNavigator = createStackNavigator({
  CreateTask: CreateTaskScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const appStackNavigator = createStackNavigator({
  Priority: PriorityScreen,
  Invitations: InvitationsScreen,
  CreateRoommateGroup: CreateRoommateGroupScreen,
  AddRoommate: AddRoommateScreen,
  Home: tabNavigator,
  Tasks: taskNavigator,
  CreateBill: CreateBillScreen,
  TaskCalendar: CalendarScreen,
  Messages: MessageScreen,
  CreateConversation: CreateConversationScreen,
  Conversation: ConversationScreen,
  Account: AccountScreen,
  ChangeDetails: ChangeDetailsScreen

},{
  initialRouteName: 'Home',
  defaultNavigationOptions: {
    header: null
  }
});

const authStack = createStackNavigator({
  Login: AuthScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const switchNav = createSwitchNavigator({
  App: appStackNavigator,
  Auth: authStack,
}, {
  initialRouteName: 'Auth',
  defaultNavigationOptions: {
    header: null
  }
});

const App = createAppContainer(switchNav);

const styles = StyleSheet.create({
  headerLeft: {
    fontSize: 20
  }
});

export default () => {
  return (
    <App />
  );
}
