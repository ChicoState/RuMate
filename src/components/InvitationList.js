import React, { useEffect, useState } from 'react';
import { View, Alert } from 'react-native';
import firebase from 'firebase';
import { ListItem } from 'react-native-elements';
import Icon from 'react-native-vector-icons/MaterialIcons'

const InvitationList = () => {
    const [invitations, setInvitations] = useState([]);
    const [clickedItemGID, setClickedItemGID] = useState("");
    const [userKey, setUserKey] = useState("");
    const [canAccept, setCanAccept] = useState(false);

    const getInvitations = () => {
        the_invitations = [];
        const invitationRef = firebase.database().ref('invites');
        invitationRef.on('value', function(snapshot) {
            let data = snapshot.val();
            for (let item in data)
            {
                if (firebase.auth().currentUser.uid == data[item].memberID)
                {
                    the_invitations.push(data[item]);
                }
            }
            setInvitations(the_invitations);
        });
    }

    const leaveGroup = (gid, iid) => {
        setClickedItemGID(gid);
        Alert.alert(
          'Already in a Group',
          'Would you like to leave your current Group?',
          [
            { text: 'Yes', onPress: () => {
              setCanAccept(true);
              firebase.database().ref('/users/' + userKey).update(
                  {
                      rid: -1
                  }
              );
            }},
            { text: 'No', onPress: () => removeInvite(gid, iid), style:'cancel'}
          ]
        )
    }

    const removeInvite = (gid, iid) => {
      Alert.alert(
        'Remove Invite',
        'Do you want to remove this invite?',
        [
          { text: 'Yes', onPress: () => {
            let data = firebase.database().ref('invites')
            data.on('value', (snapshot)=>{
              let invites = snapshot.val()
              for (invite in invites) {
                if (invites[invite].inviteID == iid) {
                  firebase.database().ref('invites/'+invite).remove();
                }
              }
            });
            getInvitations();
          }},
          { text: 'No',style:'cancel'}
        ]
      )
    }

    const joinGroup = (gid, iid) => {
      Alert.alert(
        'Accept Invitation',
        'Warning: You can only be a member of one group',
        [
          { text: 'Yes', onPress: () => {
            firebase.database().ref('/users/' + userKey).update(
                {
                    rid: gid
                }
            );
            setCanAccept(false);
            let data = firebase.database().ref('invites')
            data.on('value', (snapshot)=>{
              let invites = snapshot.val()
              for (invite in invites) {
                if (invites[invite].inviteID == iid) {
                  firebase.database().ref('invites/'+invite).remove();
                }
              }
            });
            getInvitations();
          }},
          { text: 'No', style:'cancel'}
        ]
      )
    }

    useEffect(() => {
        getInvitations();

        var users = firebase.database().ref('/users');
        users.once('value', function(snapshot) {
            snapshot.forEach(function(childSnap) {
                if (childSnap.val().uid == firebase.auth().currentUser.uid)
                {
                    setUserKey(childSnap.key);
                    if (childSnap.val().rid == -1)
                    {
                        setCanAccept(true);
                    }
                }
            })
        });
    }, []);

    return(
        <View>
            {
                invitations.map((l, i) => (
                    <ListItem
                    onPress={() => {
                      if(!canAccept)
                      {
                        leaveGroup(l.gid, l.inviteID)
                      }
                      else
                      {
                        joinGroup(l.gid, l.inviteID)
                      }
                    }}
                    key={i}
                    title={l.from}
                    rightIcon = {<Icon size={30} color='black' name="error-outline" />}
                    bottomDivider
                    />
                ))

            }

        </View>
    )
}

export default InvitationList;
