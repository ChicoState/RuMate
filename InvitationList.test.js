import React from 'react';
import renderer from 'react-test-renderer';
import firebase from 'firebase'
import App from './App'
import InvitationList from './src/components/InvitationList';

describe('<InvitationList />', () => {
  it('has 1 child', () => {
    <App />
    if (!firebase.auth().currentUser)
      firebase.auth().signInWithEmailAndPassword("messanger1@noyou.com", "password")
    const tree = renderer.create(<InvitationList />).toJSON()
    console.log(tree.children.length)
    expect(tree.children.length).toBe(1)
  });
  
  it('renders correctly', () => {
    const tree = renderer.create(<InvitationList />).toJSON()
    expect(tree).toMatchSnapshot()
  });
});