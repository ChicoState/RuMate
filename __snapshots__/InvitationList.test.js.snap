// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvitationList /> renders correctly 1`] = `
<View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 13,
                "flexDirection": "column",
                "overflow": "hidden",
                "width": 270,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(255,255,255)",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {},
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "margin": 18,
                },
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Accept Invitation
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 13,
                    "marginTop": 4,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Do you want to accept this invitation.
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 13,
                    "marginTop": 4,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Warning: You can only be a member of one group.
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderTopColor": "#A9ADAE",
                  "borderTopWidth": 0.5,
                  "flexDirection": "row",
                  "height": 46,
                  "justifyContent": "space-between",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#007ff9",
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Decline
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#A9ADAE",
                    "height": "100%",
                    "width": 0.5,
                  },
                  Object {},
                ]
              }
            />
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#007ff9",
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Accept
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
